src/services/signatureHelp.ts

```diff
@@ -525,32 +534,35 @@ function getSpreadElementCount(node: SpreadElement, checker: TypeChecker) {
     return 0;
 }
 
-function getArgumentCount(argumentsList: Node, ignoreTrailingComma: boolean, checker: TypeChecker) {
+function getArgumentCount(argumentsList: Node, checker: TypeChecker) {
     // The argument count for a list is normally the number of non-comma children it has.
     // For example, if you have "Foo(a,b)" then there will be three children of the arg
     // list 'a' '<comma>' 'b'.  So, in this case the arg count will be 2.  However, there
     // is a small subtlety.  If you have "Foo(a,)", then the child list will just have
     // 'a' '<comma>'.  So, in the case where the last child is a comma, we increase the
     // arg count by one to compensate.
-    //
-    // Note: this subtlety only applies to the last comma.  If you had "Foo(a,," then
-    // we'll have: 'a' '<comma>' '<missing>'
-    // That will give us 2 non-commas.  We then add one for the last comma, giving us an
-    // arg count of 3.
-    const listChildren = argumentsList.getChildren();
-
+    const args = argumentsList.getChildren();
     let argumentCount = 0;
-    for (const child of listChildren) {
+    let skipComma = false;
+    for (let pos = 0; pos < length(args); pos++) {

```

Comment:
Why `length(args)` instead of args.length?

Reply:
nNo particular reason really - I was probably just replicating some other code here. While `length()` has this additional ability to accept undefined inputs, it's already used by a lot of places where the input is defined. I'll replace this with plain `.length` in a second